### PR TITLE
new(ci): added gha support.

### LINKED ISSUE #292: Abandon CircleCI in favor of github actions
**Motivation**

All the falcosecurity org moved to github actions, retiring CircleCI. We should move away driverkit from CircleCI too, for consistency.
We need an arm64 runner though; either we wait for github actions to finally provide linux arm runners or we use our own test-infra nodes (like we do for Falco).

This is just to track the issue.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #292 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(ci): ported CI to gha
```

### CODE REVIEW

**Comment by FedeDP** [comment]
I disabled buildmode pie and tested that:
* i could run driverkit-amd64 on my machine
* i could run driverkit-amd64 inside an `ubuntu:18.04` docker container
* i could run driverkit-amd64 inside an `alpine:latest` docker container
* i could run driverkit-arm64 on an `ubuntu-aws` ec2 arm64 instance
* i could run driverkit-arm64 inside an arm64 `ubuntu:18.04` docker container
* i could run driverkit-arm64 inside an arm64 `alpine:latest` docker container

File: .github/workflows/ci.yml
Comment (FedeDP) [inline .github/workflows/ci.yml:1]: This is already tested on the pull request CI.

File: .github/workflows/master.yml
Comment (FedeDP) [inline .github/workflows/master.yml:1]: Master CI builds driverkit, push non-latest images and creates manifest to create multi arch docker image.

File: .github/workflows/release.yml
Comment (FedeDP) [inline .github/workflows/release.yml:1]: Release CI builds driverkit, push non-latest images and creates manifest to create multi arch docker image; finally, runs goreleaser to create the actual github release and attach release artifacts.

**Review by leogr — APPROVED** [review]

**Comment by FedeDP** [comment]
/cc @dwindsor @EXONER4TED 

If this is ok for you, i can update required checks on test-infra and we can move on with this!

**Review by EXONER4TED — DISMISSED** [review]
Looks good! Very nice 😊

**Comment by FedeDP** [comment]
Opened test-infra pr: https://github.com/falcosecurity/test-infra/pull/1329

**Comment by FedeDP** [comment]
@EXONER4TED pushed a small fix :D

**Review by EXONER4TED — APPROVED** [review]

**Pull request merged**